### PR TITLE
book/architecture: add resource index

### PR DESCRIPTION
**Pull request by hinto-janai**
### What
Adds an index section for common resources in the `Resources` section, similar to the crates index we already have.

Setting this up now and gradually editing it is probably easier than doing it all at once.

### CODE REVIEW

File: books/architecture/src/SUMMARY.md [hunk 1]
```diff
@@ -93,17 +93,20 @@
...
-        - [⚪️ The block downloader](resource-model/concurrency-and-parallelism/the-block-downloader.md)
-        - [⚪️ The verifier](resource-model/concurrency-and-parallelism/the-verifier.md)
-        - [⚪️ Thread exit](resource-model/concurrency-and-parallelism/thread-exit.md)
+- [⚪️ Resources](resources/intro.md)
+    - [⚪️ File system](resources/fs/intro.md)
+        - [🟡 Index of PATHs](resources/fs/paths.md)
+    - [⚪️ Sockets](resources/sockets/index.md)
+        - [🔴 Index of ports](resources/sockets/ports.md)
+    - [⚪️ Memory](resources/memory.md)
+    - [🟡 Concurrency and parallelism](resources/cap/intro.md)
```
Comment (hinto-janai) [inline books/architecture/src/SUMMARY.md:102]: `concurrency-and-parallelism/` is very long so -> `cap/`

File: books/architecture/src/resources/cap/threads.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# Index of threads
+This is an index of all of the system threads Cuprate actively uses.
```
Comment (hinto-janai) [inline books/architecture/src/resources/cap/threads.md:2]: Currently, is this just `global rayon` + `global tokio` + `blockchain` + `txpool`?

@Boog900 are there any threads (ephemeral/persistent) spawned in any of your code? Or any code that we call out to?

File: books/architecture/src/resources/sockets/ports.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+# Index of ports
+This is an index of all of the network sockets Cuprate actively uses.
```
Comment (hinto-janai) [inline books/architecture/src/resources/sockets/ports.md:2]: Don't think we open any right now, right? Maybe networking ones in the alpha binary?

File: books/architecture/src/resources/fs/paths.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
+# Index of PATHs
```
Comment (hinto-janai) [inline books/architecture/src/resources/fs/paths.md:1]: Do we only save database files currently?

File: books/architecture/src/resources/cap/threads.md (hunk 2 above)
Comment (Boog900) [reply books/architecture/src/resources/cap/threads.md:2]: We are going to have a thread that listens to Stdin and handles commands. Also the `blockchain` + `txpool` will use the same threadpool

File: books/architecture/src/resources/fs/paths.md (hunk 4 above)
Comment (Boog900) [reply books/architecture/src/resources/fs/paths.md:1]: we also save a `p2p_state` file, which I should add to `helper`

File: books/architecture/src/resources/sockets/ports.md (hunk 3 above)
Comment (Boog900) [reply books/architecture/src/resources/sockets/ports.md:2]: Yeah not yet, we would need #176. For the initial alpha binary I would be ok with not accepting inbound connections

**Review by Boog900 — APPROVED** [review]

**Pull request merged**